### ISSUE TITLE
rubyLibs are now broken

### ISSUE THREAD

**Issue by offlinehacker** [NixOS/nixpkgs#6340]
I appriciate work on new ruby support, but rubyLibs, are now broken. I think next time before we do this kind of changes, we should take back compatibility in consideration.

@cstrahan

**Comment by copumpkin** [comment]
Where does rubyLibs live? I can't find the string anywhere in nixpkgs.

**Comment by cstrahan** [comment]
> Where does rubyLibs live? I can't find the string anywhere in nixpkgs.

@joelteon and I phased out rubyLibs in favor of having Ruby apps use the specific versions they require per their Gemfile.

I can bring it back though, if other people have infrastructure using rubyLibs. Unless someone makes a compelling argument, I would treat that as a deprecated feature, though.

**Comment by auntieNeo** [comment]
Here is the old <a href="https://github.com/NixOS/nixpkgs/blob/65c7c61841652e070ca4f0209df142aa7c69ea73/pkgs/development/interpreters/ruby/generated.nix">ruby/generated.nix</a> file that contains a list of packages that were removed along with rubyLibs.

I have no idea how many of those have been moved to the new bundlerEnv system, but if the answer isn't "most" then I don't see how rubyLibs has been "phased out". It might be good to have rubyLibs alongside the new system for as long as packages are being <i>phased in</i>.

I was (read: still am) using at least two of those packages daily: tmuxinator and Jekyll. So far I've been surviving with a branch of the commit before they were removed (git branch rubyLibs ea5594679c438a4666f29a209f68b6cce81054f5), but it would be nice to have them back in my configuration.nix. I'm in the process of making a bundlerEnv derivation for tmuxinator based on the instructions on the mailing list, but I haven't quite figured it out yet.

Sorry, I don't want to make too much of a stink about this (I still have my rubyLibs branch), but something probably needs to happen before this hits NixOS stable and trips up even more people.

**Comment by cillianderoiste** [comment]
@cstrahan I tried to package jekyll following the instructions in the mailing list, but ran into trouble: http://comments.gmane.org/gmane.linux.distributions.nixos/15702
I use jekyll regularly at work, so it would be really great to have it back in nixpkgs.

**Comment by offlinehacker** [comment]
I'm using compass for building sass in production. As long there's no
suitable alternative for ruby gems, rubyLibs from @marcw should ne kept.
After that all the packages should be migrated.

**Comment by offlinehacker** [comment]
We already have new ruby support backported to nixos 14.12, that includes
rubyLibs, so i think master should too.

**Comment by domenkozar** [comment]
@offlinehacker can this be closed?

**Comment by cstrahan** [comment]
I have an open PR to resolve this: https://github.com/NixOS/nixpkgs/pull/6580

@offlinehacker Does that look good to merge?

**Comment by domenkozar** [comment]
@offlinehacker ping?
